src/main/groovy/org/pih/warehouse/core/identification/BlankIdentifierResolver.groovy

```diff
@@ -0,0 +1,76 @@
+package org.pih.warehouse.core.identification
+
+import groovy.util.logging.Slf4j
+import org.grails.datastore.gorm.GormEntity
+
+/**
+ * Handles generating identifiers for all instances of an entity that don't yet have one.
+ *
+ * Used by the AssignIdentifierJob as a backup retry mechanism to ensure all entities have a valid id.
+ *
+ * Classes that implement this one will almost certainly also extend from IdentifierService. The only reason this trait
+ * is kept separate from the base service is to allow services to opt out of providing the functionality if they're
+ * not called by the AssignIdentifierJob.
+ */
+@Slf4j
+trait BlankIdentifierResolver<T extends GormEntity> {
```

Comment:
This trait allows us to generalize supporting the AssignIdentifierJob that batch generates ids for rows where the id is null